.github/workflows/unit-tests.yml

```
@@ -0,0 +1,37 @@
+name: Unit tests
+
+on:
+  push:
+    branches:
+      - main
```

Since unit tests (by definition), don't reach out to outside the repo, we can run this on every PR, and not just when master is pushed.
```suggestion
on:
  pull_request:
  push:
    branches:
      - main
```